src/functional/iterator/transforms/unroll_reduce.py

```diff
@@ -1,14 +1,35 @@
+from itertools import groupby
+from typing import Iterable
+
 from eve import NodeTranslator
 from eve.utils import UIDs
 from functional.iterator import ir
 
 
+def all_equal(it: Iterable):
```

Comment:
```suggestion
def _all_equal(it: Iterable):
```
and probably put it as staticmethod into `UnrollReduce` as done with all other helper functions in this module.